packages/eslint-plugin/src/rules/dependency-checks.ts

```diff
@@ -8,7 +8,6 @@ import { satisfies } from 'semver';
 import {
   getAllDependencies,
   getPackageJson,
-  getProductionDependencies,
```

Comment:
I believe this is the only place this function is used, so we should just change it to work like `getAllDependencies` and be given package.json contents instead of reading from disk?

Either that or delete the now unused `getProductionDependencies`